tools/generator/dfg/sam/sam_device_tree.py

```diff
@@ -159,18 +183,13 @@ def _device_tree_from_properties(p):
 
         def driver_compatibility(id):
             s = id.string
-            if s.startswith("samg55"):
-                return "samg55"
-            elif s.startswith("samg54"):
-                return "samg54"
-            elif s.startswith("samg53"):
-                return "samg53"
-            elif s.startswith("samg51"):
-                return "samg51"
-            elif s.startswith("samv70"):
-                return "samv70"
-            elif s.startswith("samv71"):
-                return "samv71"
+            # TODO: do we really need this distinction?
+            # There are two groups of devices with common peripherals:
+            # - SAM x7x,G5x
+            # - SAM D09,D1x,D2x,L2x,D51,E5x
+            # TODO: What would be the appropriate naming for those groups?
+            if s.startswith("samg5"):
+                return "samg"
             else:
                 return "sam"
```

Comment:
I am not entirely sure yet what to do here. There are two separate groups of devices which have nothing in common on a register level (ignoring some IPs Atmel bought from someone else). It would be useful to be able to name them but I can't come up with anything reasonable. Any ideas?

Reply:
We could name them Samelle and Samien, which are the next names on the baby name generator right after after Sam. /s

Otherwise I would just call them `samg` and `samd`. A proper solution would need to look at all register maps and come up with a complete partitioning, so that you can see all variants and then make an informed decision on their names. Anything short of that is basically just lucky guessing.

Reply:
> Otherwise I would just call them `samg` and `samd`.

Probably the only sensible way of handling the peripheral types is doing that on a per peripherals basis like for STM32. If a certain type of UART peripheral is called `sercom` by Atmel we should use that naming instead of calling it a `samd` UART.

I have indeed looked at the register maps for many devices. For SAM it is a bit easier than for STM32. All the hardware IPs are reasonably labeled with id designators in the device data. If you find 5 different types of ADCs among all SAM devices you only have to compare those 5 to each other to find out which ones are similar enough to use the same driver.

We don't have to get all of it right from the start and can add additional refinement later on when we actually implement the device drivers in modm. We should nevertheless change the current type designators because labeling the peripherals as "samv71" for V71 devices and "samv70" for V70 makes really no sense when the hardware is 100% identical.